### PR TITLE
Auth: id response header

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Add config options and functionality to include the id of the identity making a request to all response headers.
I decided make the implementation a little more generic than what the issue describes.
I also intentionally only used fields / functions I know exist on 9.5.x because we want to backport this feature.

The response header will use the specified prefix, with default values it would be `X-Grafana-Identitity-Id` and the value will be use the namespace and id of identitiy (`user:<id>`, `api-key:<id>` or `service-account:<id>`

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/394

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by kalleep** [comment]
With this more generic option we build different headers for different identities.
* User -> `X-Grafana-User-Id`
* ApiKey -> `X-Grafana-Api-Key-Id`
* ServiceAccount -> `X-Grafana-Service-Account-Id`

I wonder if it would make more sense to use the same header for all and include the namespace in the id instead `X-Grafana-Identity-Id` would be set for all and e.g. service acccount would have id `service-account:<id>`

**Review by cinaglia — APPROVED** [review]

**Review by IevaVasiljeva — APPROVED** [review]
Nice, looks good and good shout on making it easy to backport!

I'd probably add this to the changelog, unless we want to keep quiet about this new feature. Similarly, are we planning to document this or keep this internal only for now?

**Comment by kalleep** [comment]
>Nice, looks good and good shout on making it easy to backport!
I'd probably add this to the changelog, unless we want to keep quiet about this new feature. Similarly, are we planning to document this or keep this internal only for now?

@IevaVasiljeva  I am planning to add changelog, update docs, add some tests and refactor this a bit once we have backported it :)

File: pkg/setting/setting.go [hunk 1]
```diff
@@ -1579,6 +1582,17 @@ func readAuthSettings(iniFile *ini.File, cfg *Cfg) (err error) {
 	// Azure Auth
 	AzureAuthEnabled = auth.Key("azure_auth_enabled").MustBool(false)
 	cfg.AzureAuthEnabled = AzureAuthEnabled
+
+	// ID response header
+	cfg.IDResponseHeaderEnabled = auth.Key("id_response_header_enabled").MustBool(false)
+	cfg.IDResponseHeaderPrefix = auth.Key("id_response_header_prefix").MustString("X-Grafana-")
+
+	idHeaderNamespaces := strings.Split(auth.Key("id_response_header_namespaces").MustString(""), " ")
```
Comment (gamab) [inline pkg/setting/setting.go:1590]: Nit. We can use `util.SplitString` to be consistent with how we handle lists from the ini file (supports json, or `,` and ` ` separated values )

**Review by gamab — APPROVED** [review]
LGTM! Tested and it works as a charm :)

**Pull request merged**